### ISSUE TITLE
[FR] Display chat formatting correctly

### ISSUE THREAD

**Issue by vlad2305m** [chirs241097/jline4mcdsrv#18]
Working on another project, so just dropped it here for you to copy. Can be an interesting idea :) CC0.
```java
private static String reformatMinecraftAnsi(String in) {
        return in
                .replace("§r", "\033[0m") // reset
                .replace("§l", "\033[1m") // bold
                .replace("§o", "\033[3m") // italic
                .replace("§n", "\033[4m") // underline
                .replace("§k", "\033[8m") // obfuscated
                .replace("§m", "\033[9m") // strikethrough
                .replace("§0", "\033[0;30m") // black
                .replace("§1", "\033[0;34m") // blue
                .replace("§2", "\033[0;32m") // green
                .replace("§3", "\033[0;36m") // cyan
                .replace("§4", "\033[0;31m") // red
                .replace("§5", "\033[0;35m") // purple
                .replace("§6", "\033[0;33m") // gold
                .replace("§7", "\033[0;37m") // gray
                .replace("§8", "\033[0;90m") // D grey
                .replace("§9", "\033[0;94m") // B blue
                .replace("§a", "\033[0;92m") // B green
                .replace("§b", "\033[0;96m") // B cyan
                .replace("§c", "\033[0;91m") // B red
                .replace("§d", "\033[0;95m") // B purple
                .replace("§e", "\033[0;93m") // B yellow
                .replace("§f", "\033[0;97m") // white
                ;
    }
```

**Comment by Fourmisain** [comment]
Haha, that's a nice idea, might give this one a shot!

**Comment by Fourmisain** [comment]
I was wondering if this idea could be integrated into the highlighter as well, so you could write formatted messages and have an immediate preview of them, so I gave that a try (with the wrong colors):
![test](https://user-images.githubusercontent.com/8464472/213193054-11e5a8c8-d02c-4380-b93a-40fe8ad0321c.png)

I guess black text doesn't work that well, or... too well? 😅
![black](https://user-images.githubusercontent.com/8464472/213187471-e9186233-7cfc-42c7-858d-dc6ad14ff798.png)

Not quite sure what to do about that (maybe playing with the background?)
 \- but it's technically my fault for not using an anime background image.
Code needs some work still, currently not sure how to get gray aka "bright black" as an `AttributedStyle` (**edit** got it).

**Comment by Fourmisain** [comment]
Just released a new version with the stuff in it, see f726057854807a370489bdae7d90061eae549559 for the code.

I didn't extensively test everything, had some really weird issues which I still have no idea about.
Obfuscated and strikethrough don't work for me, which is to be expected since I'm on an old Windows (though underline does work).
At first, bold and italic didn't work for me either and after some back and forth, suddenly they both worked...?
Then they stopped working again, as if my console (Cmder/ConEmu) forgot it could style text.
Tested it in C and bold works as brightening while italic doesn't do anything, which is most likely how the "vanilla" console works.
Either way, this shouldn't be an issue of the code, just of my console/system.

I ended up adding white background to the black text, just in case ... and I forgot to add that to the regular console output...
I hope it's not that important? 😅

Lemme know if it is, or if there's any other issues!
